src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java

```diff
@@ -84,12 +100,24 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
                 new BytesRestResponse(ffe.getRestStatus(), ffe.toXContent(channel.newErrorBuilder(), ToXContent.EMPTY_PARAMS))
             );
         }
+        if (!provision && !params.isEmpty()) {
+            // Consume params and content so custom exception is processed
+            params.keySet().stream().forEach(request::param);
+            request.content();
```

Comment:
The `BaseRestHandler` tracks whether parameters and content (body) are "consumed" by the RestHandler implementations.  The apparent intent of the code is to provide for some sort of "did you mean?" functionality if someone made a typo declaring a parameter.  (If you included the parameter `food` but never consumed `foo` it could be suggested as a replacement.)

Unfortunately, this check for consumed parameters overrides the returned value from the channel.  See [here](https://github.com/opensearch-project/OpenSearch/blob/247e2eec46930e2eaec970b1acfb2f7d18e3e7a9/server/src/main/java/org/opensearch/rest/BaseRestHandler.java#L106-L129) where the `action` returned from from the `prepareRequest()` method in line 1-6 isn't actually returned to the channel until line 109, while the lack of consuming params and content can throw exceptions before it gets there.  

I've known about this issue since I had to handle it in the SDK, but it reared its ugly head checking the feature flag in this repo, giving misleading error messages.  See https://github.com/opensearch-project/flow-framework/issues/513

TLDR: if you have any params in the rest path, or parse the content, and if you want your own custom message returned, you have to "consume" the params (by calling `request.param("foo")` and `content()`) before returning.

Reply:
On the same line, looking at [this](https://github.com/opensearch-project/flow-framework/blob/main/src/main/java/org/opensearch/flowframework/rest/RestProvisionWorkflowAction.java#L98), we are returning a ffe and we do have params in the rest path. Does calling `request.param` [here](https://github.com/opensearch-project/flow-framework/blob/main/src/main/java/org/opensearch/flowframework/rest/RestProvisionWorkflowAction.java#L70), helped to consume it or we need to call `request.content()` as well?

Reply:
> On the same line, looking at [this](https://github.com/opensearch-project/flow-framework/blob/main/src/main/java/org/opensearch/flowframework/rest/RestProvisionWorkflowAction.java#L98), we are returning a ffe and we do have params in the rest path. Does calling `request.param` [here](https://github.com/opensearch-project/flow-framework/blob/main/src/main/java/org/opensearch/flowframework/rest/RestProvisionWorkflowAction.java#L70), helped to consume it 

Precisely.  Currently (before this PR) the only allowable (and required) parameter for the provision API is the Workflow ID.  We "consume" the workflow ID param by calling `request.param(WORKFLOW_ID)`.  You'll notice ALL our API steps do that (see #513).  Some have a few params we need to consume.

> or we need to call `request.content()` as well?

This is similar, but just a boolean.  Prior to this PR, the provision API did not support having a body so we could rely on the `BaseRestHandler` error message, but had to at least return _something_, thus we send "null" back to the channel which gets ignored when the no-content-consumed exception is thrown.